allure-generator/src/main/java/io/qameta/allure/environment/EnvironmentPlugin.java

```diff
@@ -0,0 +1,115 @@
+        environment.withEnvironmentItems(testEnvVariables);
+        final Path envPropsFile = directory.resolve("environment.properties");
+        if (Files.exists(envPropsFile)) {
+            try (InputStream is = Files.newInputStream(envPropsFile)) {
+                final Properties properties = new Properties();
+                properties.load(is);
+                environment.withEnvironmentItems(convertItems(properties));
+            } catch (IOException e) {
+                visitor.error("Could not read environments.properties file " + envPropsFile, e);
+            }
+        }
+        final Path envXmlFile = directory.resolve("environment.xml");
+        if (Files.exists(envXmlFile)) {
+            try (FileInputStream fis = new FileInputStream(envXmlFile.toFile())) {
+                final ru.yandex.qatools.commons.model.Environment result =
+                        JAXB.unmarshal(fis, ru.yandex.qatools.commons.model.Environment.class);
+                final List<EnvironmentItem> fromXml = result.getParameter().stream()
+                        .map(param -> new EnvironmentItem().withName(param.getKey()).withValues(param.getValue()))
+                        .collect(Collectors.toList());
+                environment.withEnvironmentItems(fromXml);
+            } catch (Exception e) {
+                visitor.error("Could not read environment.xml file " + envXmlFile.toAbsolutePath(), e);
+            }
+        }
+
+        if (!environment.getEnvironmentItems().isEmpty()) {
+            visitor.visitExtra(ENVIRONMENT_BLOCK_NAME, environment);
+        }
+    }
+
+    @Override
+    public Environment getData(final Configuration configuration,
+                               final List<LaunchResults> launches) {
+        final List<Environment> launchEnvironments = launches.stream().map(launch ->
+                launch.getExtra(ENVIRONMENT_BLOCK_NAME)
+                        .filter(Environment.class::isInstance)
+                        .map(Environment.class::cast)
+                        .orElse(new Environment())).collect(Collectors.toList());
+
+        final Map<String, EnvironmentItem> globalEnvValues = new HashMap<>();
+        launchEnvironments
+                .forEach(env -> env.getEnvironmentItems()
+                        .forEach(item -> {
+                            globalEnvValues.computeIfPresent(item.getName(),
+                                (key, value) -> value.withValues(item.getValues()));
+                            globalEnvValues.putIfAbsent(item.getName(), item);
+                        }));
+        return new Environment().withEnvironmentItems(globalEnvValues.values());
+    }
+
+    @Override
+    public String getName() {
+        return "environment";
+    }
+
+    public static Collection<EnvironmentItem> convertItems(final Properties properties) {
+        return properties.keySet().stream().map(key ->
```

Comment:
`properties.stringPropertyNames()` and `Objects.toString(value)`

also you can consider using `entrySet()`

```java
.entrySet().stream()
                .map(entry -> new EnvironmentItem()
                        .withName(String.valueOf(entry.getKey()))
                        .withValue(String.valueOf(entry.getValue())))
                .collect(Collectors.toSet());
```